### ISSUE TITLE
if two monsters have the same name (as xml tag in the monstername.xml file), script wont continue

### ISSUE THREAD

**Issue by nokoji** [divinity76/YurOTSMonsterLibrary#1]

**Comment by divinity76** [comment]
i guess it throws some error? can you show me the error message?

**Comment by divinity76** [comment]
the error in question is
```
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/old_widow_queen.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/dragon_god.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/energy_devil.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/zombie.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/Gazer.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/elite_dwarf_warrior.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/the_horned_fox.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/minotaur_warrior.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/bazir.xml
processing /home/hans/projects/YurOTSMonsterLibrary/data/monster/bazir_summon.xml
PHP Fatal error:  Uncaught PDOException: SQLSTATE[23000]: Integrity constraint violation: 19 UNIQUE constraint failed: monsters.name in /home/hans/projects/YurOTSMonsterLibrary/GenerateLibrary.php:624
Stack trace:
#0 /home/hans/projects/YurOTSMonsterLibrary/GenerateLibrary.php(624): PDOStatement->execute()
#1 /home/hans/projects/YurOTSMonsterLibrary/GenerateLibrary.php(649): {closure}()
#2 /home/hans/projects/YurOTSMonsterLibrary/GenerateLibrary.php(17): getDB()
#3 {main}
  thrown in /home/hans/projects/YurOTSMonsterLibrary/GenerateLibrary.php on line 624
hans@xDevAd:~/projects/YurOTSMonsterLibrary$ 

```
hmm yeah.. a "proper" fix would be to make a dedicated monster page for each monster with the same name, but then i would have to come up with a "demon_1" "demon_2" "demon_..." scheme, which would make generating proper urls more difficult too.... hmmm, can't be arsed doing a proper fix now,

as a QUICKFIX im just going to do
```php
            try {
                $stm->execute();
            } catch (\Throwable $ex) {
                if (false !== strpos($ex->getMessage(), 'UNIQUE constraint failed: monsters.name')) {
                    // multiple monsters with the same name in monsters.xml ?
                    // FIXME: we should create a dedicated monster page for each monster with the same name,
                    // but as a quickfix i will just keep the first monster and ignore the rest of the monsters with the same name =/
                    continue;
                }
                throw $ex;
            }
```

**Comment by divinity76** [comment]
quickfix is implemented in 7e49e772863b67ad22bb4215b25555d4c3c336b1 
but.. a proper fix with a dedicated page for each monster with duplicate name would still be preferable =/
